src/OrchardCore.Modules/OrchardCore.Layers/Services/LayerFilter.cs

```diff
@@ -113,13 +113,14 @@ public async Task OnResultExecutionAsync(ResultExecutingContext context, ResultE
                     bool display;
                     if (!layersCache.TryGetValue(layer.Name, out display))
                     {
-                        if (String.IsNullOrEmpty(layer.Rule))
+                        var task = _ruleService.EvaluateAsync(layer.LayerRule);
+                        if (!task.IsCompletedSuccessfully)
                         {
-                            display = false;
+                            display = AwaitedRule(task).Result;
```

Comment:
Normally this kind of pattern is used to not have to use an await and return a Task / ValueTask directly, but here we already are in an async task, here the call is synchronous as with `Getawaiter().GetResult()`.